src/app/components/pages/IsaacCompetition/EntryForm/CompetitionEntryForm.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from "react";
+import { Form, Row, Col, Container, FormGroup, Label, Input } from "reactstrap";
+import { InputType } from "reactstrap/es/Input";
+import { AppGroup } from "../../../../../IsaacAppTypes";
+import { isaacApi, useAppSelector, useAppDispatch, AppDispatch } from "../../../../state";
+import { selectors } from "../../../../state/selectors";
+import { SchoolInput } from "../../../elements/inputs/SchoolInput";
+import { api } from "../../../../services";
+import { showErrorToast, showSuccessToast } from "../../../../state/actions/popups";
+
+const COMPETITON_ID = "isaac_competition_25";
+interface CompetitionEntryFormProps {
+  handleTermsClick: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+}
+
+const CompetitionEntryForm = ({ handleTermsClick }: CompetitionEntryFormProps) => {
```

Comment:
while this is quite a large file, it's really well made. Especially the `renderFormGroup` usage. Really good work. 

My only suggestion would be if there's any way to pull some of the logic out in this file to a supporting utility file it could reduce the size of it somewhat.